### PR TITLE
CSV cleaning

### PR DESCRIPTION
**Pull request by kofimeighan**
Created processor.py to clean general csv data, and updated requirements.txt

Examples of the cleaned data:
Screenshot 2020-07-13 at 5 35 09 PM

### CODE REVIEW

File: portfolio/src/main/data_proccessing/processor.py [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+for csv_column in csv_columns:
+    stripped_csv_column = csv_column.lower().replace("'", "")
+    for column_of_interest in COLUMNS_OF_INTEREST:
+        if column_of_interest.lower() in stripped_csv_column:
+            columns_to_keep.append(csv_column)
+            break
 
-print("python_script works")
+df = df[columns_to_keep]    
+df.to_csv('clean_data/police_killings_2013.csv', index = False)
```
Comment (briafassler9) [inline portfolio/src/main/data_proccessing/processor.py]: Add newline at end of file

File: portfolio/src/main/data_proccessing/processor.py [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+
 import pandas as pd
+import numpy as np
+
+# TODO(kofimeighan): Create a function that can iterate over all files within 
+# the raw_data directory
+COLUMNS_OF_INTEREST = ['name', 'race','date', 'address', 'city',
+                     'state', 'zipcode',]
+PUNCTUATION = '''!()-[];:'"\,<>./?@#$%^&*_~'''
+df = pd.read_csv('raw_data/police_killings_2013.csv', index_col = 0)
```
Comment (briafassler9) [inline portfolio/src/main/data_proccessing/processor.py]: What is df supposed to represent?

File: portfolio/src/main/data_proccessing/processor.py (hunk 1 above)
Comment (kofimeighan) [reply portfolio/src/main/data_proccessing/processor.py]: There is a newline, github just doesn't show it i guess?

image

File: portfolio/src/main/data_proccessing/processor.py (hunk 2 above)
Comment (kofimeighan) [reply portfolio/src/main/data_proccessing/processor.py]: df is shorthand for dataframe which is how i saw a lot of documentation refer to it as, would readability be better if I changed it to dataframe?

Comment (Jaxobc) [reply portfolio/src/main/data_proccessing/processor.py]: You can give the dataframe a more descriptive name.

**Review by Jaxobc — APPROVED** [review]

**Comment by Jaxobc** [comment]
> Created processor.py to clean general csv data, and updated requirements.txt
> 
> Examples of the cleaned data:
> Screenshot 2020-07-13 at 5 35 09 PM

This looks great Kofi! Could you also show what the data looked like before you cleaned it. 
(Edit) Perhaps you do not need the images and could just include the files that contain the cleaned data and the raw data.

File: portfolio/src/main/data_proccessing/processor.py [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+processor.py is used to clean csv's and create data suitable to be
+stored within a database."""
+
 import pandas as pd
+import numpy as np
+
+# TODO(kofimeighan): Create a function that can iterate over all files within 
+# the raw_data directory
+COLUMNS_OF_INTEREST = ['name', 'race','date', 'address', 'city',
+                     'state', 'zipcode',]
```
Comment (wangtong-g) [inline portfolio/src/main/data_proccessing/processor.py]: Please align 'state' with 'name'.

File: portfolio/src/main/data_proccessing/processor.py [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+df = pd.read_csv('raw_data/police_killings_2013.csv', index_col = 0)
```
Comment (wangtong-g) [inline portfolio/src/main/data_proccessing/processor.py]: Shouldn't you commit csv file too?

Comment (wangtong-g) [reply portfolio/src/main/data_proccessing/processor.py]: Also, what's the data in "raw_data/police_killings_2013.csv" look like?

File: portfolio/src/main/data_proccessing/processor.py (hunk 1 above)
Comment (briafassler9) [reply portfolio/src/main/data_proccessing/processor.py]: Just checked with each other, all good.

File: portfolio/src/main/data_proccessing/processor.py (hunk 2 above)
Comment (briafassler9) [reply portfolio/src/main/data_proccessing/processor.py]: +1

File: portfolio/src/main/data_proccessing/processor.py (hunk 3 above)
Comment (Jaxobc) [reply portfolio/src/main/data_proccessing/processor.py]: There is an extra comma after zipcode

**Comment by kofimeighan** [comment]
Sample of what the raw csv file looks like: 
image

**Review by briafassler9 — APPROVED** [review]

File: portfolio/src/main/data_proccessing/processor.py [hunk 5]
```diff
@@ -1,3 +1,35 @@
...
+
+processor.py is used to clean csv's and create data suitable to be
+stored within a database."""
+
 import pandas as pd
+import numpy as np
+
+# TODO(kofimeighan): Create a function that can iterate over all files within 
+# the raw_data directory
+COLUMNS_OF_INTEREST = ['name', 'race','date', 'address', 'city',
```
Comment (wangtong-g) [inline portfolio/src/main/data_proccessing/processor.py]: Nit: one space before 'date'.

File: portfolio/src/main/data_proccessing/processor.py [hunk 6]
```diff
@@ -1,3 +1,35 @@
...
+                       'state', 'zipcode',]
+dataframe = pd.read_csv('raw_data/police_killings_2013.csv', index_col = 0)
+csv_columns = dataframe.columns
+columns_to_keep = []
+
+# TODO(kofimeighan): save the zipcode as an int not a long within the database
+for csv_column in csv_columns:
+    stripped_csv_column = csv_column.lower().replace("'", "")
```
Comment (wangtong-g) [inline portfolio/src/main/data_proccessing/processor.py:30]: This seems unnecessary.

File: portfolio/src/main/data_proccessing/processor.py [hunk 7]
```diff
@@ -1,3 +1,35 @@
...
+    for column_of_interest in COLUMNS_OF_INTEREST:
```
Comment (wangtong-g) [inline portfolio/src/main/data_proccessing/processor.py:31]: Any reason we don't remove the ineligible rows, for example, the ones have "Unknown race" or "Name withheld by police"?

One step further, which of the columns in "COLUMNS_OF_INTEREST" are mandatory? We should only keep the rows that have values for mandatory columns.

File: portfolio/src/main/data_proccessing/processor.py [hunk 8]
```diff
@@ -1,3 +1,35 @@
...
+        if column_of_interest.lower() in stripped_csv_column:
+            columns_to_keep.append(csv_column)
+            break
 
-print("python_script works")
+dataframe = dataframe[columns_to_keep]    
+dataframe.to_csv('clean_data/police_killings_2013.csv', index = False)
```
Comment (kennzoid) [inline portfolio/src/main/data_proccessing/processor.py]: nit: no spaces around equal sign

File: portfolio/src/main/data_proccessing/processor.py [hunk 9]
```diff
@@ -1,3 +1,35 @@
...
+stored within a database."""
+
 import pandas as pd
+import numpy as np
+
+# TODO(kofimeighan): Create a function that can iterate over all files within 
+# the raw_data directory
+COLUMNS_OF_INTEREST = ['name', 'race','date', 'address', 'city',
+                       'state', 'zipcode',]
+dataframe = pd.read_csv('raw_data/police_killings_2013.csv', index_col = 0)
```
Comment (kennzoid) [inline portfolio/src/main/data_proccessing/processor.py]: 'raw_data/police_killings_2013.csv' should go in a constant at the top.

and nit: no space around the equal signs in function calls

**Review by kennzoid — APPROVED** [review]

File: portfolio/src/main/data_proccessing/processor.py (hunk 6 above)
Comment (kofimeighan) [reply portfolio/src/main/data_proccessing/processor.py:30]: I do this incase a csv has a field with mismatched names. Ex. Date's or Zipcode's.

File: portfolio/src/main/data_proccessing/processor.py (hunk 7 above)
Comment (kofimeighan) [reply portfolio/src/main/data_proccessing/processor.py:31]: No reason, I believe that it something I will implement next. 

Everything within COLUMN_OF_INTEREST is an essential field for at least Jacob, Bria, or I's features.

File: portfolio/src/main/data_proccessing/processor.py (hunk 6 above)
Comment (wangtong-g) [reply portfolio/src/main/data_proccessing/processor.py:30]: I am pretty sure "date in Date's" is still true in python

Comment (kofimeighan) [reply portfolio/src/main/data_proccessing/processor.py:30]: you are correct. I'm going to remove it now.

**Pull request merged**